mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -721,6 +727,29 @@ def LLVM_DbgValueOp : LLVM_DbgIntrOp<"dbg.value", "value",
     LLVM_DILocalVariableAttr:$varInfo,
     DefaultValuedAttr<LLVM_DIExpressionAttr, "{}">:$locationExpr
   );
+  let llvmBuilder = [{
+    // Debug intrinsics without debug locations are invalid.
+    if(!builder.getCurrentDebugLocation())
+      return success();
+    llvm::Function *parentFn = builder.GetInsertBlock()->getParent();
+    llvm::DILocalScope *scope;
+    if (auto scopeLoc =
+            opInst.getLoc()->findInstanceOf<FusedLocWith<LLVM::DIScopeAttr>>())
+      scope = llvm::cast<llvm::DILocalScope>(
+          moduleTranslation.translateDebugInfo(scopeLoc.getMetadata()));
+    else
+      scope = parentFn->getSubprogram();
+
+    llvm::Module *module = builder.GetInsertBlock()->getModule();
+    llvm::DIBuilder DB(*module);
```

Comment:
```suggestion
    llvm::DIBuilder debugInfoBuilder(*module);
```
nit: local variables should start with a lower case letter. Let's maybe use debugInfoBuilder or similar?